cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -108,41 +90,45 @@
 
   @GET
   @Path("/**/metadata/subparts")
-  public void getSubparts(HttpRequest request, HttpResponder responder) throws NotFoundException, BadRequestException {
+  public void getSubparts(HttpRequest request, HttpResponder responder) throws BadRequestException {
     //TODO(Rohit): Get the sub keys here and return that. This is the new API which we will support for browsing
   }
 
   @GET
   @Path("/**/metadata")
   public void getMetadata(HttpRequest request, HttpResponder responder,
-                          @QueryParam("scope") String scope) throws NotFoundException, BadRequestException {
-    MetadataEntity metadataEntity = getMetadataEntityFromPath(request.uri(), "/metadata");
-    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(getMetadata(metadataEntity, scope),
+                          @QueryParam("scope") String scope, @QueryParam("type") @DefaultValue("") String type)
```

Comment:
Why do you need a defaultValue? Isn't it null if it's not in the URL? I am not sure what the empty string helps with...